src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -77,9 +77,6 @@ def trace(self) -> JinjaTrace:
             except IndexError:
                 pos2 = len(trace_template_output)
             p = trace_template_output[pos1 + 1 : pos2]
-            is_set_or_macro = p[:3] == "set"
-            if is_set_or_macro:
-                p = p[3:]
```

Comment:
With this change, `set` and macro blocks no longer generate trace output like this, thus this code can be removed.